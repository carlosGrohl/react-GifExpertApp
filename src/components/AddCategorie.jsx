
import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategorie = ({onNewCategory}) => {

    const [inputValue, setinputValue] = useState('');

    const onInputChange = (event) => {
        //console.log(event.target.value);
        setinputValue(event.target.value);
    };

    const onSubmit = ( event ) => {
        event.preventDefault();
        // si la funcion tiene menos de 2 caracteres hace un return
        if (inputValue.trim().length <=1) return;
        
        onNewCategory(inputValue.trim());
        setinputValue('');
    };

  return (
    <form onSubmit={ (event) => onSubmit(event) }  aria-label='form'>
        <input
            id='buscador'
            name='Buscador'
            type="text"
            placeholder="Buscar gifs"
            value={ inputValue }
            //onChange={(event) => onInputChange(event) }
            onChange={ onInputChange }
        />
    </form>
        
  )
}

//Obligamos que la funcion onNewCategory sea requerida
AddCategorie.propTypes ={
    onNewCategory: PropTypes.func.isRequired,
}