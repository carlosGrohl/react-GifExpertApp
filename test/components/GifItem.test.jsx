import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem />', () => { 
    test(' Debe de hacer match con el snapshot', () => {
        const title = 'Dragon ball';
        const url = 'https://i.pinimg.com/originals/c4/17/62/c41762a3049ef0fc129651997389e389.png';
        
        const { container } = render( <GifItem title={ title } url={ url }/>);
        expect(container).toMatchSnapshot();
    });
 })