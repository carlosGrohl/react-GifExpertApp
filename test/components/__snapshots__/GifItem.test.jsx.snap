// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem />  Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon ball"
      src="https://i.pinimg.com/originals/c4/17/62/c41762a3049ef0fc129651997389e389.png"
    />
    <p>
       
      Dragon ball
       
    </p>
  </div>
</div>
`;
